### DISCUSSION TITLE
handling compressed files

### DISCUSSION THREAD

**Discussion by bdnbje** [mhx/dwarfs#121] [Ideas]
This is probably out of scope of this project but since its coming around being a very good archiver can a way to recompress zlib streams on the fly be possible to be a added feature in this project. zlib is the most common compression algorithm used in files png pdf a lot of generic data used in a regular file system uses zlib streams if there could be a way to handle them on the fly could make dwarfs state of the art. Theres already some projects that do handle this task like [https://github.com/deus-libri/preflate](preflate), [https://github.com/schnaader/precomp-cpp](precomp) and [https://github.com/google/grittibanzli](grittibanzli)

**Comment by mhx** [comment]
Yeah, this idea has come up before, see #109
